### PR TITLE
Implement get file in clinical document service AB#13872

### PR DESCRIPTION
**Pull request by rayking29**
# Fixes or Implements [AB#13872](https://dev.azure.com/qslvic/304a1f8c-dace-4f85-adf3-bf563d5b3a39/_workitems/edit/13872)

## Description

Implement get file in clinical document service [AB#13872](https://dev.azure.com/qslvic/304a1f8c-dace-4f85-adf3-bf563d5b3a39/_workitems/edit/13872)
Add model for PhsaClinicalDocumentRecord [AB#13913](https://dev.azure.com/qslvic/304a1f8c-dace-4f85-adf3-bf563d5b3a39/_workitems/edit/13913)
Code cleanup

## Testing

- [ ] Unit Tests Updated
- [ ] Functional Tests Updated
- [x] Not Required

## UI Changes

Screen Shot 2022-09-14 at 10 45 01 AM

## Items to Review:

-   [General PR Guidelines](https://github.com/bcgov/healthgateway/wiki/PRguidance)

### CODE REVIEW

File: Apps/ClinicalDocument/src/Api/IClinicalDocumentsApi.cs [hunk 1]
```diff
@@ -28,9 +28,18 @@ public interface IClinicalDocumentsApi
         /// <summary>
         /// Retrieves clinical document records by patient identifier.
         /// </summary>
-        /// <param name="pid">The pid to get clinical document records.</param>
+        /// <param name="pid">The patient id to get clinical document records.</param>
```
Comment (sslaws) [inline Apps/ClinicalDocument/src/Api/IClinicalDocumentsApi.cs:31]: Nit: Are we sure we know that pid = patient id and not personal id?

File: Apps/ClinicalDocument/src/Models/ClinicalDocumentRecord.cs [hunk 2]
```diff
@@ -24,15 +24,15 @@ namespace HealthGateway.ClinicalDocument.Models
     public class ClinicalDocumentRecord
     {
         /// <summary>
-        /// Gets or sets the ID.
+        /// Gets or sets the id.
         /// </summary>
-        [JsonPropertyName("healthDataId")]
+        [JsonPropertyName("id")]
```
Comment (sslaws) [inline Apps/ClinicalDocument/src/Models/ClinicalDocumentRecord.cs:29]: Nit: I don't think we typically use mapping annotations for our models.

**Review by sslaws — DISMISSED** [review]

File: Apps/ClinicalDocument/src/Api/IClinicalDocumentsApi.cs (hunk 1 above)
Comment (rayking29) [reply Apps/ClinicalDocument/src/Api/IClinicalDocumentsApi.cs:31]: Yes according to the PatientIdentity model:

Screen Shot 2022-09-14 at 11 17 04 AM

File: Apps/ClinicalDocument/src/Models/ClinicalDocumentRecord.cs (hunk 2 above)
Comment (rayking29) [reply Apps/ClinicalDocument/src/Models/ClinicalDocumentRecord.cs:29]: Many of them do, the ones I used as examples. Is this something we are avoiding now?

Comment (MikeLyttle) [reply Apps/ClinicalDocument/src/Models/ClinicalDocumentRecord.cs:29]: I think Stephen is referring to the models used solely to transfer data between our front-end and back-end. In which case, that's on me for putting them in there and not noticing.

Comment (rayking29) [reply Apps/ClinicalDocument/src/Models/ClinicalDocumentRecord.cs:29]: Okay I will remove those.

File: Apps/ClinicalDocument/src/Services/ClinicalDocumentService.cs [hunk 3]
```diff
@@ -67,13 +73,14 @@ public async Task<RequestResult<IEnumerable<ClinicalDocumentRecord>>> GetRecords
...
+                IApiResponse<PhsaHealthDataResponse> apiResponse =
+                    await this.clinicalDocumentsApi.GetClinicalDocumentRecords(response.ResourcePayload?.PatientIdentity?.Pid.ToString()).ConfigureAwait(true);
                 if (apiResponse.IsSuccessStatusCode)
                 {
+                    IList<ClinicalDocumentRecord> clinicalDocuments = this.autoMapper.Map<IList<ClinicalDocumentRecord>>(apiResponse.Content?.Data);
                     requestResult.ResultStatus = ResultType.Success;
-                    requestResult.ResourcePayload = apiResponse.Content;
-                    requestResult.TotalResultCount = apiResponse.Content.Count();
+                    requestResult.ResourcePayload = clinicalDocuments;
+                    requestResult.TotalResultCount = apiResponse.Content?.Data.Count();
```
Comment (MikeLyttle) [inline Apps/ClinicalDocument/src/Services/ClinicalDocumentService.cs:83]: Nit: using `clinicalDocuments.Count` here would remove the need for the null-conditional operator.

**Review by BrianMaki — APPROVED** [review]

File: Apps/ClinicalDocument/src/Services/ClinicalDocumentService.cs [hunk 4]
```diff
@@ -106,7 +113,24 @@ public async Task<RequestResult<EncodedMedia>> GetFileAsync(string hdid, string
             RequestResult<PersonalAccount?> response = await this.personalAccountsService.GetPatientAccountAsync(hdid).ConfigureAwait(true);
             if (response.ResultStatus == ResultType.Success)
             {
+                string? pid = response.ResourcePayload?.PatientIdentity?.Pid.ToString();
                 this.logger.LogDebug("PID Fetched: {Pid}", response.ResourcePayload?.PatientIdentity?.Pid);
```
Comment (MikeLyttle) [inline Apps/ClinicalDocument/src/Services/ClinicalDocumentService.cs:117]: Nit: you can reference the variable that was just defined to make the log statement a tad cleaner.

**Review by MikeLyttle — APPROVED** [review]

File: Apps/ClinicalDocument/src/Services/ClinicalDocumentService.cs (hunk 4 above)
Comment (rayking29) [reply Apps/ClinicalDocument/src/Services/ClinicalDocumentService.cs:117]: Yeah looks like I started to do that then lost it. I'll fix that up with my unit tests.

File: Apps/ClinicalDocument/src/Services/ClinicalDocumentService.cs (hunk 3 above)
Comment (rayking29) [reply Apps/ClinicalDocument/src/Services/ClinicalDocumentService.cs:83]: Good point, I will fix that with my unit tests.

**Pull request merged**